### PR TITLE
ddl: remove temporary testing methods

### PR DESCRIPTION
**Pull request by tisonkun**
### What problem does this PR solve?

Issue Number: ref #27180

```release-note
None
```

### CODE REVIEW

File: ddl/db_partition_test.go [hunk 1]
```diff
@@ -3670,40 +3670,6 @@ func TestAddPartitionReplicaBiggerThanTiFlashStores(t *testing.T) {
 	require.Equal(t, "[ddl:-1]DDL job rollback, error msg: [ddl] add partition wait for tiflash replica to complete", err.Error())
 }
 
-func TestDropAndTruncatePartition(t *testing.T) {
-	// Useless, but is required to initialize the global infoSync
```
Comment (tisonkun) [inline ddl/db_partition_test.go:3674]: We don't need these wrappers, just directly test the inner functions. Also `TestCreateTables` doesn't test the exact functionality.

**Comment by hawkingrei** [comment]
@tisonkun Have some useless function

**Comment by tisonkun** [comment]
@hawkingrei removed.

**Review by tiancaiamao — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**